corehq/apps/reports/views.py

```diff
@@ -181,19 +184,21 @@ def _inner(request, domain, *args, **kwargs):
                     user_allowed = True
 
                 if not user_allowed:
-                    raise Http403()
+                    return HttpResponseForbidden()
 
                 return view_func(request, domain, *args, **kwargs)
-            return require_form_view_permission(view_func)(request, domain, *args, **kwargs)
 
-        if login_decorator:
-            return login_decorator(_inner)
-        else:
-            return _inner
+            try:
+                response = require_form_view_permission(view_func)(request, domain, *args, **kwargs)
+            except PermissionDenied:
+                response = HttpResponseForbidden()
```

Comment:
What's this about?  Does `PermissionDenied` return something weird for API requests?  I assumed that's what's currently returned when `require_form_view_permission` fails, just curious what prompted the change.

Reply:
Good question. 

I noticed when `require_form_view_permission` fails, it doesn't return a response but rather raises the `PermissionDenied` error. This made it a little harder to test, so I thought I'd explicitly catch it and return a 403. I think this makes for a better user experience also (I think if an exception is unhandled the user might see the 500 page?).

Reply:
That's a special exception that Django catches in middleware and uses to display a 403 page, though I think you're right here in that the 403 page isn't a super appropriate response for an API (JSON would be better).  We have a `json_error` decorator that does the same thing you do here, and I see that `api/resources/auth.py` does it in a few places as well.  Sounds good to me.

For what it's worth, in tests you can also do something like this:

https://github.com/dimagi/commcare-hq/blob/master/corehq/apps/reports/tests/test_views.py#L388-L389